### ISSUE TITLE
Can support for declaration from init.pp be added?

### ISSUE THREAD

**Issue by jcpunk** [crayfishx/puppet-purge#4]
It would simplify my workflow if I could pass an argument to the class for creating the type.

Something like:
```
class purge (
  $purges = {}
) {
  if $purges (
    validate_hash($purges)
    create_resources(xxxxx)
  }
```

The parameter probably needs a much better name.....

Similar behavior in Puppet Approved yo61-logrotate

**Comment by crayfishx** [comment]
@jcpunk certainly not against it - I would prefer to use Puppet 4.x/ 3 + future parser and do it with an iterator rather than use `create_resources` though - would that still work  for you?

**Comment by jcpunk** [comment]
Yeah, that should be fine

**Comment by crayfishx** [comment]
@jcpunk Feel free to put in a PR if you fancy tackling this yourself, something along the lines of `purge::resources` I think would be better, `purge::purges` sounds a bit weird.  And using an iterator  rather than `create_resources`.... if not I'll knock something up this week.

Cheers

**Comment by crayfishx** [comment]
@jcpunk This feature is added in #8 and I'll be pushing this in 1.2.0 - hopefully later today... let me know if that feature works for you.

**Comment by jcpunk** [comment]
Looks great!

**Comment by crayfishx** [comment]
@jcpunk This feature has been released in 1.2.0, there are some other changes to, see the CHANGELOG.    Thanks.
